src/traces/treemap/style.js

```diff
@@ -35,37 +35,57 @@ function styleOne(s, pt, trace, opts) {
     var ptNumber = cdi.i;
     var lineColor;
     var lineWidth;
-    var opacity;
-
-    var depthFade = function(n) {
-        var base = trace.marker.opacitybase;
-        var step = trace.marker.opacitystep;
-
-        return n === 0 ? base :
-            Math.max(0, Math.min(1, n * step));
-    };
+    var fillColor = cdi.color;
+    var isRoot = helpers.isHierarchyRoot(pt);
+    var opacity = 1;
 
     if(hovered) {
         lineColor = trace._hovered.marker.line.color;
         lineWidth = trace._hovered.marker.line.width;
-        opacity = trace._hovered.marker.opacity;
     } else {
-        if(helpers.isHierarchyRoot(pt)) {
+        if(isRoot && fillColor === 'rgba(0,0,0,0)') {
```

Comment:
This solution correctly handle the default root case that sets `cdi.color` to `rgba(0,0,0,0)` here:

https://github.com/plotly/plotly.js/blob/1dff468da972eb9417c8b3c690e9accfafdf30eb/src/traces/sunburst/calc.js#L269-L270

but, we'll probably need to extend it all other `rgba` colors with a `0` alpha channel - as the current depthfade algorithm gives a white `fillColor`.